### ISSUE TITLE
Parse fail for Cisco IOS BGP: "next-hop-self all"

### ISSUE THREAD

**Issue by saparikh** [batfish/batfish#532]
Failed to parse lhr-border-02.cfg:
    org.batfish.common.BatfishException: Error parsing configuration file: '/Users/samir/git/batfish/containers/samir/testrigs/tr_5e851bc1-1673-4eb2-b852-8e2bad30dc77/testrig/configs/lhr-border-02.cfg'
       at org.batfish.job.ParseVendorConfigurationJob.call(ParseVendorConfigurationJob.java:311)
       at org.batfish.job.ParseVendorConfigurationJob.call(ParseVendorConfigurationJob.java:32)
       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
       at java.lang.Thread.run(Thread.java:748)
    Caused by: org.batfish.main.ParserBatfishException: Parser error
       at org.batfish.main.Batfish.parse(Batfish.java:388)
       at org.batfish.job.ParseVendorConfigurationJob.call(ParseVendorConfigurationJob.java:300)
       ... 5 more
    Caused by: org.batfish.common.DebugBatfishException: 
    parser: CiscoParser: line 143:35: extraneous input 'all' expecting NEWLINE
    Offending Token:  line 143:35  ALL:'all'  <== mode:DEFAULT_MODE
    Error parsing top (leftmost) parser rule in stack: '[next_hop_self_bgp_tail bgp_tail neighbor_flat_rb_stanza router_bgp_stanza_tail router_bgp_stanza stanza cisco_configuration]'.
    Parse tree of current rule:
    (next_hop_self_bgp_tail
      NEXT_HOP_SELF:'next-hop-self'  <== mode:DEFAULT_MODE)
    Unconsumed tokens:
     line 143:35  ALL:'all'  <== mode:DEFAULT_MODE
     line 143:38  NEWLINE:'\n'  <== mode:DEFAULT_MODE
    Previous 10 tokens:
     line 142:30 (HIDDEN)  WS:' '  <== mode:DEFAULT_MODE
     line 142:31  DEC:'65331'  <== mode:DEFAULT_MODE
     line 142:36  NEWLINE:'\n'  <== mode:DEFAULT_MODE
     line 143:0 (HIDDEN)  WS:' '  <== mode:DEFAULT_MODE
     line 143:1  NEIGHBOR:'neighbor'  <== mode:DEFAULT_MODE
     line 143:9 (HIDDEN)  M_NEIGHBOR_WS:' '  <== mode:M_NEIGHBOR
     line 143:10  IP_ADDRESS:'10.10.30.2'  <== mode:M_NEIGHBOR
     line 143:20 (HIDDEN)  WS:' '  <== mode:DEFAULT_MODE
     line 143:21  NEXT_HOP_SELF:'next-hop-self'  <== mode:DEFAULT_MODE
     line 143:34 (HIDDEN)  WS:' '  <== mode:DEFAULT_MODE
    Parser state info:
    _cadant: false
    _disableUnrecognized: false
    _multilineBgpNeighbors: false
    
    Error context lines:
       133:     bgp router-id 10.10.255.8
       134:     bgp log-neighbor-changes
       135:     bgp bestpath compare-routerid
       136:     neighbor FW peer-group
       137:     neighbor 10.10.10.37 remote-as 65201
       138:     neighbor 10.10.10.37 peer-group FW
       139:     neighbor 10.10.10.37 route-map SET_LOCAL_PREF in
       140:     neighbor 10.10.10.45 remote-as 65202
       141:     neighbor 10.10.10.45 peer-group FW
       142:     neighbor 10.10.30.2 remote-as 65331
    >>>143:     neighbor 10.10.30.2 next-hop-self all
       144:    !
       145:    ip forward-protocol nd
       146:    !
       147:    !
       148:    no ip http server
       149:    no ip http secure-server
       150:    !
       151:    ip access-list extended MATCH_ALL_BGP
       152:     permit ip any any
       153:    !
    
       at org.batfish.grammar.BatfishParserErrorListener.syntaxError(BatfishParserErrorListener.java:121)
       at org.batfish.grammar.BatfishParserErrorListener.syntaxError(BatfishParserErrorListener.java:136)
       at org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41)
       at org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:544)
       at org.antlr.v4.runtime.DefaultErrorStrategy.reportUnwantedToken(DefaultErrorStrategy.java:349)
       at org.antlr.v4.runtime.DefaultErrorStrategy.singleTokenDeletion(DefaultErrorStrategy.java:513)
       at org.antlr.v4.runtime.DefaultErrorStrategy.recoverInline(DefaultErrorStrategy.java:439)
       at org.antlr.v4.runtime.Parser.match(Parser.java:206)
       at org.batfish.grammar.cisco.CiscoParser.next_hop_self_bgp_tail(CiscoParser.java)
       at org.batfish.grammar.cisco.CiscoParser.bgp_tail(CiscoParser.java)
       at org.batfish.grammar.cisco.CiscoParser.neighbor_flat_rb_stanza(CiscoParser.java)
       at org.batfish.grammar.cisco.CiscoParser.router_bgp_stanza_tail(CiscoParser.java)
       at org.batfish.grammar.cisco.CiscoParser.router_bgp_stanza(CiscoParser.java)
       at org.batfish.grammar.cisco.CiscoParser.stanza(CiscoParser.java:34571)
       at org.batfish.grammar.cisco.CiscoParser.cisco_configuration(CiscoParser.java:4027)
       at org.batfish.grammar.cisco.CiscoCombinedParser.parse(CiscoCombinedParser.java:42)
       at org.batfish.grammar.cisco.CiscoCombinedParser.parse(CiscoCombinedParser.java:9)
       at org.batfish.main.Batfish.parse(Batfish.java:386)
       ... 6 more
STATISTICS
  Total warnings:
    Red Flag: 6
  Parsing results:
    Parsed successfully: 10
    Contained unrecognized line(s): 6
    Failed to parse: 2

**Comment by dhalperi** [comment]
[template message] Now that we have parser recovery, I'm closing a bunch of older "this doesn't parse" bugs. I'm assuming that filers will ping those that are important to them.

**Comment by dhalperi** [comment]
[template message] Now that we have parser recovery, I'm closing a bunch of older "this doesn't parse" bugs. I'm assuming that filers will ping those that are important to them.

**Comment by saparikh** [comment]
if the issue isn't fixed, then why close it?

**Comment by dhalperi** [comment]
Parser no longer fails on unrecognized lines. If there's a real use case someone is blocked on, file a new issue for implementing this support (among the many potential new features to add).
